public/angular/js/controllers/reports/show.js

```diff
@@ -1,141 +1,205 @@
-    };
-    $scope.unlinkIncident = function() {
-      $scope.report._incident = '';
-      Report.update({ id: $scope.report._id }, $scope.report);
-    };
-
-    $scope.markAsRead = function (report) {
-      if (report.read) return;
-      report.read = true;
-      Report.save({id: report._id}, report);
-    };
-
-    $scope.tagsToString = Tags.tagsToString;
-    $scope.markAsRead(data.report);
-
-    $scope.updateReport = function(report) {
-      angular.extend($scope.report, report);
-      if (!$scope.incidentsById[report._incident]) {
-        Incident.get({ id: report._incident }, function(inc) {
-          incidents.results.push(inc);
-          $scope.incidentsById[report._incident] = inc;
+    '$state',
+    '$scope',
+    '$stateParams',
+    '$window',
+    'mediaOptions',
+    'FlashService',
+    'data',
+    'incidents',
+    'Report',
+    'Incident',
+    'incidents',
+    'Tags',
+    'smtcTags',
+    'comments',
+    'Socket',
+    'Queue',
+    'sources',
+    'paginationOptions',
+    '$translate',
+    function($rootScope, $state, $scope, $stateParams, $window, mediaOptions, flash, data, incidents, Report, Incident,
+             incidents, Tags, smtcTags, comments, Socket, Queue, sources, paginationOptions,  $translate) {
+      $scope.incidents = incidents.results;
+      $scope.incidentsById = {};
+      $scope.currentPath = $rootScope.$state.current.name;
+      $scope.smtcTags = smtcTags;
+      $scope.smtcTagsById = {};
+      $scope.comments = comments.results;
+      $scope.visibleComments = new Queue(paginationOptions.perPage);
+      $scope.sources = sources;
+      $scope.sourcesById = {};
+      $scope.pagination = {
+        page: parseInt($stateParams.page) || 1,
+        total: comments.total,
+        visibleTotal: comments.total,
+        perPage: paginationOptions.perPage,
+        start: 0,
+        end: 0
+      };
+
+      var init = function() {
+        $scope.commentsById = $scope.comments.reduce(groupById, {});
+        $scope.sourcesById = $scope.sources.reduce(groupById, {});
+        $scope.incidentsById = $scope.incidents.reduce(groupById, {});
+        $scope.smtcTagsById = $scope.smtcTags.reduce(groupById, {});
+
+        Socket.on('stats', updateStats);
+        Socket.join('stats');
+        if ($scope.currentPath === 'report') {
+          Socket.join('reports');
+          Socket.on('report:updated', $scope.updateReport.bind($scope));
+        }
+        var visibleComments = paginate($scope.comments);
+        $scope.visibleComments.addMany(visibleComments);
+
+      };
+      var updateStats = function(stats) {
+        $scope.stats = stats;
+      };
+
+      /**
+       * Saves a front-end report to the back end.
+       * @param {Report} report
+       */
+      $scope.saveReport = function(report) {
+        Report.save({ id: report._id }, report, function() {
+        }, function() {
+          flash.setAlertNow("Sorry, but that report couldn't be saved.");
```

Comment:
Would be lovely to at least log an error to the console, lest it be swallowed by the void.